### ISSUE TITLE
[BUG] Training using init-model gives normal lcurve and bad model

### ISSUE THREAD

**Issue by zjgemi** [deepmodeling/deepmd-kit#3751]
### Bug summary

In the DPGen [workflow](https://workflows.deepmodeling.com/workflows/argo/dpa20w-rmwk9?tab=workflow), training in iter-0 seems all right. The model trained in iter-1 (with init-model) has a large RMSE ~100meV, while the lcurve shows a better accuracy
![20240506-193625](https://github.com/deepmodeling/deepmd-kit/assets/27884723/0f3e1bcf-5617-46d4-9fda-8e789be31b07)
For the worst system, the RMSE increase by a factor >30 after training of iter-1.
![20240506-194007](https://github.com/deepmodeling/deepmd-kit/assets/27884723/16619218-6830-47c1-be1c-2c62777309d6)
This phenomenon does not appear when using finetune (instead of init-model) in iter-1.

### DeePMD-kit Version

stable-0411

### Backend and its version

Pytorch

### How did you download the software?

docker

### Input Files, Running Commands, Error Log, etc.

iter1input.zip

### Steps to Reproduce

bash aefcb166ade9f2faf80a15e8a6f0d0cb70a6d33a.sub

**Comment by Chengqian-Zhang** [comment]
I followed the steps of
1. fine-tuning based on the multitask pre-trained model
2. init-model based on the finetuning model obtained in step1

and found that I am able to reproduce the bug on the `stable_0411` branch, but everything works well on the latest `devel` branch, so you can test if it's still an issue on the latest `devel` branch.
